.github/workflows/ci-action.yml

```diff
@@ -19,6 +19,12 @@ jobs:
         with:
             path: ouroboros_repo
             submodules: recursive
+      - name: Checkout Spark-Config
```

Comment:
Minor: I think it's better to add this to the pyproject dependencies via
```toml
dependences = [
  "spark_config @ git+https://github.com/MIT-SPARK/Spark-Config@main"
]
```
though not sure how it'll play with CI given git isn't actually always available

Reply:
Ah, yeah I'll give that a try. I'm pretty sure it should work.

Reply:
@nathanhhughes Actually do you know how specifying the dependency this way works if you do it in multiple packages? If multiple packages specify this does it work as long as they refer to the same version?

Reply:
Yeah, you can specify the version in addition to the branch at the end of the url (setup tools has documentation somewhere), but in general it just always installs main every time it gets pulled in as a dependency